### PR TITLE
Search contact role

### LINKED ISSUE #17: Search Contacts by Role
GET /api/v1/contacts?role=ROLE

Search entire contact database for all matches to role. Should NOT return matching contacts if they are already assigned to an active incident

### LINKED ISSUE #18: Get all Contacts related to an Incident
GET /api/v1/incidents/:incident_id/contacts?role=ROLE

Returns all contacts part of a specific incident and their assigned role in that incident

### PR DESCRIPTION
**Pull request by amsmyth1**
### What Changed?
- Complete happy path for returning Contacts related to an incident
- Complete happy path for searching contacts based on role and not assigned to an active project
- Create poro to return specialized attributes for IncidentContacts (like specific title to incident, distance in minutes and miles)
### What's Next?
- REFACTOR AND CLEAN UP. I definitely rushed this to get to back end. I'd love to spend time tonight cleaning it up. I want to refactor all incident contact related items to use a PORO
### Known Issues?
- There are old incident-contact items that need to utilize the poro
### Test Coverage?
- Is 95% and will be addressed in the next PR
### README & Documentation
- Endpoints updated for search and returning all contacts for an incident
### Closed Issues
Close #17 
Close #18

### CODE REVIEW

File: app/controllers/api/v1/contacts_controller.rb [hunk 1]
```diff
@@ -13,6 +13,9 @@ def create
         role = Role.find_or_create_by(title: role)
         ContactRole.create(contact_id: contact.id, role_id: role.id)
       end
+      role = Role.find_or_create_by(title: contact_params[:job_title])
+      ContactRole.create(contact_id: contact.id, role_id: role.id)
```
Comment (klatour324) [inline app/controllers/api/v1/contacts_controller.rb:17]: I think I understand what is going on but I am not 100% positive. So after we create the contact, we go through each role where we find that role based on the title, and then we create a ContactRole once the role is defined for each given role. Then after that we are redefining the role based the contact_params passed in and create a whole new ContactRole with that information correct?

File: app/poros/incident_contact_info.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    @name = attributes[:name]
+    @email = attributes[:email]
+    @phone_number = attributes[:phone_number]
+    @incident_location = attributes[:incident_location]
+    @contact_location = attributes[:contact_location]
+    @distance_miles = calculate_distance_miles
+    @distance_minutes = calculate_distance_minutes
+  end
+
+  def calculate_distance_miles
```
Comment (jmichlitsch) [inline app/poros/incident_contact_info.rb:16]: I like how you were able to put place holders for our distance micro-service to mock our data coming in.

File: app/models/contact.rb [hunk 3]
```diff
@@ -16,6 +16,19 @@ def assigned_to_active_incident?
...
+    joins(:incidents).where("incidents.active = ?", true)
+  end
+
+  def self.find_by_role(role_query)
+    select('contacts.*')
+    .joins(:contact_roles)
+    .select("contact_roles.contact_id")
+    .joins(:roles)
+    .select('roles.title')
+    .where("lower(roles.title) LIKE ?", "%#{role_query.downcase}%")
```
Comment (klatour324) [inline app/models/contact.rb:32]: Nice work on the ActiveRecord here!

**Review by jmichlitsch — APPROVED** [review]
I like how you were able to mock some of the data for our distance microservice and were able to incorporate the incident contact csv files to work fairly effortlessly

**Review by klatour324 — APPROVED** [review]
Nice work here! I just had a question that was more of a clarification for my own understanding of how it works, but otherwise this looks really good!

File: app/controllers/api/v1/contacts_controller.rb (hunk 1 above)
Comment (amsmyth1) [reply app/controllers/api/v1/contacts_controller.rb:17]: correct! that is exactly how it is functioning

**Pull request merged**